back/app/models/identity.rb

```diff
@@ -25,22 +25,27 @@ class Identity < ApplicationRecord
 
   validates :uid, :provider, presence: true
 
-  def self.find_with_omniauth(auth)
-    find_by(uid: auth['uid'], provider: auth['provider'])
-  end
+  class << self
+    def find_or_build_with_omniauth(auth, authver_method)
+      auth_to_persist = if authver_method.respond_to?(:filter_auth_to_persist)
+        authver_method.filter_auth_to_persist(auth)
+      else
+        auth
+      end
```

Comment:
I suggest to do the same as for `profile_to_uid` and have a base implementation in the parent class.
```suggestion
      auth_to_persist = authver_method.filter_auth_to_persist(auth)
```